pkg/parse/run.go

```diff
@@ -247,6 +247,21 @@ func run(ctx context.Context, p Parser, trigger string, state *reconcilerState)
 	// pull the source commit and directory with retries within 5 minutes.
 	gs.Commit, syncDir, gs.Errs = hydrate.SourceCommitAndDirWithRetry(util.SourceRetryBackoff, opts.SourceType, opts.SourceDir, opts.SyncDir, opts.ReconcilerName)
 
+	// Add pre-sync annotations to the object.
+	// If updating the object fails, it's likely due to a signature verification error
+	// from the webhook. In this case, add the error as a source error.
+	if gs.Errs == nil {
+		var err error
+		if opts.SourceType == "git" {
+			err = p.resetSourceAnnotations(ctx)
```

Comment:
Suggest merging `resetSourceAnnotations` and `setSourceAnnotations`.

The consolidated `setSourceAnnotations` function should
1. always set the `source-commit` annotation.
2. Set the `image-url` to empty for Git sources, and set to `p.Options.SourceRepo` for OCI and Helm.

Reply:
For future-proofing, let's rename `image-url` to `source-url`, but still keep it unset for the Git source.

Reply:
Done with the latest patch, the source-commit is no longer removed when switching back to git. Modified the tests to reflect that.